### PR TITLE
React login components

### PR DESCRIPTION
**Pull request by czmud**
added use reducer for login form
JSON web tokens now working for authentication

### CODE REVIEW

File: client/src/components/forms/LoginForm.tsx [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+        ...oneOrganizer,
+        [action.type]: action.payload
+    };
+}
+
+const LoginForm = () => {
+    const navigate = useNavigate();
+    const [oneOrganizer, dispatch] = React.useReducer( reducer, new LoginOrganizer());
+    const [showPassword, setShowPassword] = React.useState<boolean>(false);
+    const [ errors, setErrors ] = React.useState<LoginErrors[]>([]);
```
Comment (czmud) [inline client/src/components/forms/LoginForm.tsx:45]: Login errors are all parsed from the error response and will be in the format of:
```js
errors = [
    {
        path: "fieldName",
        error: "error text"
    },
    {
        path: "fieldName",
        error: "error text"
    },
]
```
these should be good to go for incorporating into the login form

File: client/src/components/buttons/LogoutButton.tsx [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    const redirectAfterSuccessfulLogout = () => {
+        navigate("/login");
+    }
+
+    const logOrganizerOut = (event: React.MouseEvent<HTMLButtonElement>, successCallback: Function) => {
+        event.preventDefault();
+        axios.post("http://localhost:8000/api/organizers/logout")
+            .then(() => successCallback())
+            .catch(errors => console.log(errors));
+    }
```
Comment (czmud) [inline client/src/components/buttons/LogoutButton.tsx:17]: Logout button submits a post request using axios. then once successful logout response is return, it executes callback function to redirect to main login page.

We may want to change this to redirect to main "/events" route on logout instead now that I think about it.

Feel free to delete this button once functionality is incorporated into logout button in header bar.

File: client/src/views/Dashboard.tsx [hunk 3]
```diff
@@ -86,6 +87,7 @@ const Dashboard = () => {
         </TableBody>
         </Table>
     </TableContainer>
+    <LogoutButton></LogoutButton>
```
Comment (czmud) [inline client/src/views/Dashboard.tsx:90]: put example logout button on dashboard for testing

**Review by czmud — COMMENTED** [review]
leaving notes

File: client/src/components/buttons/LogoutButton.tsx (hunk 2 above)
Comment (GreenJ84) [reply client/src/components/buttons/LogoutButton.tsx:17]: Perfect will do, Guest display sounds like the better route

File: client/src/components/forms/LoginForm.tsx (hunk 1 above)
Comment (GreenJ84) [reply client/src/components/forms/LoginForm.tsx:45]: Will do after EventForm data paths set-up

File: client/src/views/Dashboard.tsx (hunk 3 above)
Comment (GreenJ84) [reply client/src/views/Dashboard.tsx:90]: Depending on button build, I might keep the component and update props to add into HeaderBar -> will update you on final decisions

**Review by GreenJ84 — APPROVED** [review]
All changes look good to merge! Next step implementations will be done today.

**Pull request merged**